crates/chain/tests/test_indexed_tx_graph.rs

```diff
@@ -452,3 +452,264 @@ fn test_list_owned_txouts() {
         );
     }
 }
+
+/// Check conflicts between two in mempool transactions. Tx with older `seen_at` is filtered out.
+#[test]
+fn test_unconfirmed_conflicts() {
+    let (local_chain, mut graph, desc) = common::single_descriptor_setup();
+    let mut spk_iter = SpkIterator::new(&desc);
+    let (parent_tx, tx_conflict_1, tx_conflict_2) = common::setup_conflicts(&mut spk_iter);
+
+    // Parent tx is confirmed at height 2.
+    let _ = graph.insert_relevant_txs(
+        [&parent_tx].iter().map(|tx| {
+            (
+                *tx,
+                [ConfirmationHeightAnchor {
+                    anchor_block: (2, *local_chain.blocks().get(&2).unwrap()).into(),
+                    confirmation_height: 2,
+                }],
+            )
+        }),
+        None,
+    );
+
+    // Conflict 1 is seen at 100
+    let _ = graph.insert_relevant_txs([&tx_conflict_1].iter().map(|tx| (*tx, None)), Some(100));
+
+    // Conflict 2 is seen at 200
+    let _ = graph.insert_relevant_txs([&tx_conflict_2].iter().map(|tx| (*tx, None)), Some(200));
+
+    let txout_confirmations = graph
+        .list_owned_txouts(&local_chain, local_chain.tip().unwrap())
+        .map(|txout| txout.chain_position)
+        .collect::<BTreeSet<_>>();
+
+    let mut utxos = graph
+        .list_owned_unspents(&local_chain, local_chain.tip().unwrap())
+        .collect::<Vec<_>>();
+
+    // We only have 2 txouts. The confirmed `tx_1` and latest `tx_conflict_2`
+    assert_eq!(txout_confirmations.len(), 2);
+    assert_eq!(
+        txout_confirmations,
+        [
+            ObservedAs::Confirmed(ConfirmationHeightAnchor {
+                anchor_block: (2u32, *local_chain.blocks().get(&2).unwrap()).into(),
+                confirmation_height: 2
+            }),
+            ObservedAs::Unconfirmed(200)
+        ]
+        .into()
+    );
+
+    // We only have one utxo. The latest `tx_conflict_2`.
+    assert_eq!(
+        utxos.pop().unwrap().chain_position,
+        ObservedAs::Unconfirmed(200)
+    );
```

Comment:
Why could we not `assert` all of `utxo`?

Reply:
We can, but will make the tests code too verbose. The point here is to test various txout positions. If we wanna test the whole structure of `FullTxOut`, could be done in a separate test.